test/test_cuda.py

```diff
@@ -292,6 +292,26 @@ def test_memory_allocation(self):
                 torch.cuda.caching_allocator_delete(mem)
                 self.assertEqual(torch.cuda.memory_allocated(), prev)
 
+    def test_get_alloc_free_events(self):
+        gc.collect()
```

Comment:
I added it in case some workers don't fully clean up, so this could free some memory before the allocation? I can remove it if you think it is not necessary